src/UseCases/FeeChange/FeeChangeUseCase.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\MembershipContext\UseCases\FeeChange;
+
+use WMDE\Fundraising\MembershipContext\Domain\FeeChangeException;
+use WMDE\Fundraising\MembershipContext\Domain\Model\FeeChange;
+use WMDE\Fundraising\MembershipContext\Domain\Model\FeeChangeState;
+use WMDE\Fundraising\MembershipContext\Domain\Repositories\FeeChangeRepository;
+use WMDE\Fundraising\MembershipContext\Infrastructure\PaymentServiceFactory;
+use WMDE\Fundraising\MembershipContext\UseCases\ApplyForMembership\ApplicantType;
+use WMDE\Fundraising\MembershipContext\UseCases\ApplyForMembership\ApplicationValidationResult;
+use WMDE\Fundraising\PaymentContext\Domain\UrlGenerator\DomainSpecificContext;
+use WMDE\Fundraising\PaymentContext\Services\URLAuthenticator;
+use WMDE\Fundraising\PaymentContext\UseCases\CreatePayment\FailureResponse;
+use WMDE\Fundraising\PaymentContext\UseCases\CreatePayment\PaymentCreationRequest;
+use WMDE\Fundraising\PaymentContext\UseCases\CreatePayment\PaymentParameters;
+
+class FeeChangeUseCase {
+
+	public function __construct(
+		private readonly FeeChangeRepository $feeChangeRepository,
+		private readonly PaymentServiceFactory $paymentServiceFactory,
+		private readonly URLAuthenticator $urlAuthenticator,
+		private readonly ShowFeeChangePresenter $presenter
+	) {
+	}
+
+	public function showFeeChange( string $uuid ): void {
```

Comment:
maybe we could call this `showFeeChangeForm` to make it less abstract in the classes which use it in other contexts